Mathlib/CategoryTheory/Square.lean

```diff
@@ -0,0 +1,250 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Comma.Arrow
+import Mathlib.CategoryTheory.CommSq
+
+/-!
+# The category of commutative squares
+
+In this file, we define a bundled version of `CommSq`
+which allows to consider commutative squares as
+objects in a category `Square C`.
+
+The four objects in a commutative square are
+numbered as follows:
+```
+X₁ ⟶ X₂
+|    |
+v    v
+X₃ ⟶ X₄
```

Comment:
This diagram looks a bit off in both github and my vscode. Maybe use `->` arrows to avoid kerning issues?